crates/polars-plan/src/plans/aexpr/utils.rs

```diff
@@ -1,121 +1,144 @@
-use bitflags::bitflags;
-
 use super::*;
 
-fn has_series_or_range(ae: &AExpr) -> bool {
-    matches!(
-        ae,
-        AExpr::Literal(LiteralValue::Series(_) | LiteralValue::Range { .. })
-    )
-}
+/// Checks if the top-level expression node is elementwise. If this is the case, then `stack` will
+/// be extended further with any nested expression nodes.
+pub fn is_elementwise(stack: &mut Vec<Node>, ae: &AExpr, expr_arena: &Arena<AExpr>) -> bool {
```

Comment:
Base `is_elementwise()` function - recursion is done through a nodes `stack` which allows the caller to inspect the exprs.